### ISSUE TITLE
Zone Listing Selection Dropdown Doesn't List All Zones

### ISSUE THREAD

**Issue by BIueRich** [workingconcept/cloudflare-craft-plugin#72]
### Bug Report

If you have a Cloudflare Account with numerous zones (we have about 80), the dropdown selection once verifying your API credentials, does not show all zones, and repeats some zones. Or most likely confusing some domain name pairings with other zones ids and incorrectly listing them by name.

Additionally, if you don't verify credentials first and simply add in the zone ID into the text field and save, it generates an internal error and Admin is no longer accessible. We had to remove plugin to get admin working again.

### Plugin Version

putyourlightson/craft-cloudflare:^2.1.0

### Craft CMS Version

4.10.0

### PHP Version

8.1

**Comment by bencroker** [comment]
> If you have a Cloudflare Account with numerous zones (we have about 80), the dropdown selection once verifying your API credentials, does not show all zones, and repeats some zones.

The zones are output as retrieved from Cloudflare’s API, which you can test using CURL as follows:

```sh
curl --request GET \
  --url https://api.cloudflare.com/client/v4/zones \
  --header 'X-Auth-Email: test@email.com' \
  --header 'X-Auth-Key: 1234567890' \
  --header 'Content-Type: application/json'
```

> Additionally, if you don't verify credentials first and simply add in the zone ID into the text field and save, it generates an internal error and Admin is no longer accessible. We had to remove plugin to get admin working again.

I’ve addressed this issue in https://github.com/putyourlightson/craft-cloudflare/commit/0151146a6019398ee42250c76cc74933a1323400, which should prevent the error from causing a timeout. If you continue to get errors then please check the `storage/logs/web-*.log` files for specific errors.

**Comment by BIueRich** [comment]
Hi there.. Thanks for the update. A few things I discovered however.
First, if I verify credentials in the plugin for the API, the dropdown will list a max of 50 zones. We have just over 100 I think, but certainly this exact number seems to be that the results are being limited.

Second, I noticed that the internal server error if I tried to save the settings was a result of CKEditor Plugin not having settings (disabling this plugin allowed the Cloudflare plugin settings to save).

I can see that it did save the zone ID I specified in my project.yaml file, but due to the above 50 zone limitation, it still doesn't show the zone selected in the dropdown.

Thanks

**Comment by bencroker** [comment]
> First, if I verify credentials in the plugin for the API, the dropdown will list a max of 50 zones.

Pagination wasn’t working correctly and has been fixed in https://github.com/putyourlightson/craft-cloudflare/commit/d6e0cf4a284befa5e5b8a3f579d1212151f6a946, please let me know if that resolves it for you.

> Second, I noticed that the internal server error if I tried to save the settings was a result of CKEditor Plugin not having settings (disabling this plugin allowed the Cloudflare plugin settings to save).

That sounds unusual but thanks for the heads-up.

**Comment by BIueRich** [comment]
That fixed it! Thanks for your work.
As FYI, this is what seemed to conflict or show in the logs when I got internal errors, which I had not actually setup CKEditor and maybe was the cause for ANY plugin trying to save the project.yaml file. Once disabling CKEditor, Cloudflare plugin saved.

2024-06-17 12:49:35 [web.WARNING] [application] Attempting to set settings on a plugin that doesn't have settings: ckeditor

**Comment by bencroker** [comment]
Great! Released in [2.1.1](https://github.com/putyourlightson/craft-cloudflare/releases/tag/2.1.1) and [3.0.1](https://github.com/putyourlightson/craft-cloudflare/releases/tag/3.0.1).
